### ISSUE TITLE
Smart Playlists, both AND and OR

### ISSUE THREAD

**Issue by sabret00the** [clementine-player/Clementine#4744]
I should be able to create a Smart Playlist with tracks with a 4* or more rating AND the genre to include either 'folk' OR 'indie'.

**Comment by desputinski** [comment]
Yep, this would be a great feature. Amarok 1.4 had it and I used it alot.

**Comment by Moonbase59** [comment]
I strongly support that having AND/OR combinations would be just great. Here’s a sample:
I often play rock music from a certain decade and would really wish for smart playlists like "70s Rock", "80s Rock" and so forth.

Currently, I have to select the "All Titles" smart playlist and do an akward playlist search like this one:
`((originalyear:>=1980 originalyear:<=1989) OR comment:1980s) genre:rock -genre:pop`

**Comment by desputinski** [comment]
@Moonbase59: In my case the scenario is that I would like to combine Genres: Triphop OR Pop OR Jazz with AND Rating greater than 4. Right now this is not possible in Clementine.
For the problem you are having my solution is a workaround that I have different genres for "pop-70s", "pop-80s" and so on. But this is not optimal...

**Comment by ITicDigger** [comment]
I support the importance of adding complex queries with parentheses and the ability to use AND and OR in a single query. I keep track labels in the comment tag. I just list them separated by commas. When I need a set of tracks with certain tags I make a smart playlist, but with the system that is now in Clementine, I can only make simple queries. For example, if I need to get the tracks with tags "fun", "dark", but without the tag "melancholy" - I have nothing happens in this player. I have to use another player for such requests and it's frustrating, because in my opinion Clementine is the best player on Linux.

**Comment by ITicDigger** [comment]
> `((originalyear:>=1980 originalyear:<=1989) OR comment:1980s) (grouping:rock OR genre:rock) -genre:pop`
Wow! And its works! Where i can see full reference of syntax such queries?
